identity/src/main/java/com/android/identity/SessionEncryptionReader.java

```diff
@@ -126,7 +126,7 @@ private PublicKey deviceEngagementExtractEDeviceKey(byte[] encodedDeviceEngageme
         if (!(cipherSuiteDataItem instanceof Number)) {
             throw new IllegalArgumentException("Cipher suite not a Number");
         }
-        int cipherSuite = ((Number) cipherSuiteDataItem).getValue().intValue();
+        long cipherSuite = ((Number) cipherSuiteDataItem).getValue().longValue();
```

Comment:
same here, should use a helper method that checks for overflow for the conversion